client/src/resources/map/MapView.js

```diff
@@ -1,39 +1,63 @@
 import React from 'react';
-import { GoogleMapLoader, GoogleMap, Marker } from 'react-google-maps';
+import { GoogleMapLoader, GoogleMap, Marker, Rectangle, InfoWindow } from 'react-google-maps';
 import { connection } from './MapModel.js';
-import AddMarkerButton from './AddMarkerButton';
 
 class MapView extends React.Component {
 
   constructor(props) {
     super(props);
-    this.onMapClick = this.onMapClick.bind(this);
-    this.onMarkerRightclick = this.onMarkerRightclick.bind(this);
+    this.state = {
+      sectors: [],
+    };
+    this.onSectorMouseover = this.onSectorMouseover.bind(this);
+    this.sectors = [];
+    this.sectorColor = '#3d3d3d';
+    this.sector = 'Center Sector!';
+    this.sectorPosition = { lat: 37.754862, lng: -122.431558 };
   }
 
   componentDidMount() {
     this.props.populateMarkers();
+    this.createSectorsForMap();
   }
 
-  onMapClick(event) {
-    if (this.props.addMarkerToMapState) {
-      this.props.addMarker(
-        {
-          key: this.props.markers.length,
-          position: event.latLng,
-          defaultAnimation: 2
-        }
-      )
-      this.props.addMarkerToMapStateSwitch();
-      this.props.switchUploadModalState();
-    }
+  onSectorMouseover(index, position) {
+    this.sector = `The sector at this location is: ${index}`;
+    this.sectorPosition = position;
   }
 
-  onMarkerRightclick (index) {
-    this.props.removeMarker(index);
+  createSectorsForMap() {
+    fetch('/artPlace/max/rank', {
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: {
+        Accept: 'application/json',
+        ContentType: 'application/json',
+      },
+    }).then(response => response.json()).then(body => {
+      // using the id i should get back from body, I'd make
+      // another call to max rank at the id the square is located at
+      // then using the user id i get back from that route, ill grab
+      // the user color to save in the sectorColor for each individual
+      // sector
+      const LAT_LONG_TRUNCATE = 3;
```

Comment:
Since the markers are to precision 4, I have to have the sectors surround the markers. Also, they seem like decent sizes on the map.